src/java/main/org/apache/zookeeper/server/FinalRequestProcessor.java

```diff
@@ -460,11 +465,20 @@ public void processRequest(Request request) {
             if (request.type == OpCode.closeSession) {
                 cnxn.sendCloseSession();
             }
+            checkLatency(request);
         } catch (IOException e) {
             LOG.error("FIXMSG",e);
         }
     }
 
+    private void checkLatency(Request request) {
+        long requestLatency = Time.currentElapsedTime() - request.createTime;
+        if((requestWarnThresholdMs > -1 && requestLatency > requestWarnThresholdMs) || requestWarnThresholdMs == 0) {
+            LOG.warn("Request "  + request + " exceeded threshold. Took " + requestLatency + " ms.");
```

Comment:
You can leverage the use of interpolation of Strings in logging:
```
LOG.warn("Request {} exceeded {}", obj1, obj2);
```

Again, such a minor suggestion that it's up to you adopt it or not. Cheers!